test/parallel/test-fs-watch-enoent.js

```diff
@@ -1,21 +1,64 @@
 'use strict';
+
+// This verifies the error thrown by fs.watch.
+
 const common = require('../common');
 const assert = require('assert');
 const fs = require('fs');
+const tmpdir = require('../common/tmpdir');
+const path = require('path');
+const nonexistentFile = path.join(tmpdir.path, 'non-existent');
+const uv = process.binding('uv');
+
+tmpdir.refresh();
+
+{
+  const validateError = (err) => {
+    assert.strictEqual(nonexistentFile, err.path);
+    assert.strictEqual(nonexistentFile, err.filename);
+    assert.strictEqual(err.syscall, 'watch');
+    if (err.code === 'ENOENT') {
+      assert.strictEqual(
+        err.message,
+        `ENOENT: no such file or directory, watch '${nonexistentFile}'`);
+      assert.strictEqual(err.errno, uv.UV_ENOENT);
+      assert.strictEqual(err.code, 'ENOENT');
+    } else {  // AIX
+      assert.strictEqual(
+        err.message,
+        `ENODEV: no such device, watch '${nonexistentFile}'`);
+      assert.strictEqual(err.errno, uv.UV_ENODEV);
+      assert.strictEqual(err.code, 'ENODEV');
```

Comment:
@nodejs/platform-aix is this expected behaviour?

Reply:
```patch
--- a/test/parallel/test-fs-watch-enoent.js
+++ b/test/parallel/test-fs-watch-enoent.js
@@ -6,6 +6,7 @@ const fs = require('fs');
 assert.throws(function() {
   fs.watch('non-existent-file');
 }, function(err) {
+  console.log(err)
   assert(err);
   assert(/non-existent-file/.test(err));
   assert.strictEqual(err.filename, 'non-existent-file');
```

with a console.log in the callback of the existing code, this is what I get in AIX:

```trace
{ Error: watch non-existent-file ENODEV
    at FSWatcher.start (fs.js:1346:19)
    at Object.fs.watch (fs.js:1369:11)
    at /tmp/gireesh/node/test/parallel/test-fs-watch-enoent.js:7:6
    at getActual (assert.js:421:5)
    at Function.throws (assert.js:430:18)
    at Object.<anonymous> (/tmp/gireesh/node/test/parallel/test-fs-watch-enoent.js:6:8)
    at Module._compile (module.js:670:30)
    at Object.Module._extensions..js (module.js:681:10)
    at Module.load (module.js:581:32)
    at tryModuleLoad (module.js:521:12)
  errno: 'ENODEV',
  code: 'ENODEV',
  syscall: 'watch non-existent-file',
  filename: 'non-existent-file' }
```
hope this helps.

Reply:
I just tested with the changes from this PR on a local AIX box, and this test looks good - the error messages are matching, thanks!